packages/node-core/src/integrations/http/httpServerIntegration.ts

```diff
@@ -0,0 +1,456 @@
+   * Read more about Release Health: https://docs.sentry.io/product/releases/health/
+   *
+   * Defaults to `true`.
+   */
+  sessions?: boolean;
+
+  /**
+   * Number of milliseconds until sessions tracked with `trackIncomingRequestsAsSessions` will be flushed as a session aggregate.
+   *
+   * Defaults to `60000` (60s).
+   */
+  sessionFlushingDelayMS?: number;
+
+  /**
+   * Do not capture the request body for incoming HTTP requests to URLs where the given callback returns `true`.
+   * This can be useful for long running requests where the body is not needed and we want to avoid capturing it.
+   *
+   * @param url Contains the entire URL, including query string (if any), protocol, host, etc. of the incoming request.
+   * @param request Contains the {@type RequestOptions} object used to make the incoming request.
+   */
+  ignoreRequestBody?: (url: string, request: RequestOptions) => boolean;
+
+  /**
+   * Controls the maximum size of incoming HTTP request bodies attached to events.
+   *
+   * Available options:
+   * - 'none': No request bodies will be attached
+   * - 'small': Request bodies up to 1,000 bytes will be attached
+   * - 'medium': Request bodies up to 10,000 bytes will be attached (default)
+   * - 'always': Request bodies will always be attached
+   *
+   * Note that even with 'always' setting, bodies exceeding 1MB will never be attached
+   * for performance and security reasons.
+   *
+   * @default 'medium'
+   */
+  maxRequestBodySize?: 'none' | 'small' | 'medium' | 'always';
+}
+
+const _httpServerIntegration = ((options: HttpServerIntegrationOptions = {}) => {
+  const _options = {
+    sessions: options.sessions ?? true,
+    sessionFlushingDelayMS: options.sessionFlushingDelayMS ?? 60_000,
+    maxRequestBodySize: options.maxRequestBodySize ?? 'medium',
+    ignoreRequestBody: options.ignoreRequestBody,
+  };
+
+  return {
+    name: 'HttpServer',
+    setupOnce() {
+      const onHttpServerRequestStart = ((_data: unknown) => {
+        const data = _data as { server: Server };
+
+        instrumentServer(data.server, _options);
+      }) satisfies ChannelListener;
+
+      subscribe('http.server.request.start', onHttpServerRequestStart);
+    },
+
+    afterAllSetup(client) {
+      if (DEBUG_BUILD && client.getIntegrationByName('Http')) {
+        debug.warn(
+          'It seems that you have manually added `httpServerIntegration` while `httpIntegration` is also present. Make sure to remove `httpServerIntegration` when adding `httpIntegration`.',
+        );
+      }
+    },
+  };
+}) satisfies IntegrationFn;
+
+/**
+ * This integration handles request isolation, trace continuation and other core Sentry functionality around incoming http requests
+ * handled via the node `http` module.
+ */
+export const httpServerIntegration = _httpServerIntegration as (
```

Comment:
exporting the types that we depend on here directly so we can call this in a type-safe way.